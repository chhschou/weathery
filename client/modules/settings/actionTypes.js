export const UPDATE_LOCATION_ID = 'settings/UPDATE_LOCATION_ID'
export const UPDATE_ISCELSIUS = 'settings/UPDATE_ISCELSIUS'