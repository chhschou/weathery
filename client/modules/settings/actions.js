import * as actionTypes from 'modules/settings/actionTypes'

export function updateCurrentLocationId (locationId) {
  return {type: actionTypes.UPDATE_CURRENT_LOCATION_ID, locationId}
}