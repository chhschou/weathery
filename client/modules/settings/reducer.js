import * as actionTypes from './actionTypes'

const initialState = {
  isCelsius: true,
  currentLocationId: -1
}

export default function (state = initialState, action) {
  switch (action.type) {
    case actionTypes.UPDATE_ISCELSIUS:
      return {...state, isCelsius: action.isCelsius} 
    case actionTypes.UPDATE_LOCATION_ID:
      return {...state, currentLocationId: action.currentLocationId}
    default:
      return state
  }
}