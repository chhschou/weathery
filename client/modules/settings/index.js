import * as actions from './actions'
import reducer from './reducer'
import * as actionTypes from './actionTypes'

export default { reducer, actionTypes, actions }