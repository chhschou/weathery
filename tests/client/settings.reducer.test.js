import 'dotenv/config'
import settings from 'modules/settings'

test('settings has correct initial state', () => {
  const nextState = settings.reducer(undefined, {})
  expect(nextState).toMatchSnapshot()
})

test('current location id updates correctly', () => {
  const expected = 2
  const action = {
    type: settings.actionTypes.UPDATE_CURRENT_LOCATION_ID,
    locationId: expected
  }

  const nextState = settings.reducer(undefined, action)
  expect(nextState.currentLocationId).toBe(expected)
})

test('isCelsius updates correctly', () => {
  const expected = false
  const action = {
    type: settings.actionTypes.UPDATE_ISCELSIUS,
    isCelsius: expected
  }

  const nextState = settings.reducer(undefined, action)
  expect(nextState.isCelsius).toBe(expected)
})