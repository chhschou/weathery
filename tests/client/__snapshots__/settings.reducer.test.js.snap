// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`settings has correct initial state 1`] = `
Object {
  "currentLocationId": -1,
  "geocodeApiKey": "AIzaSyByY6eucpVZ6TJuVx8wAj4DDkUbsIFvf7s",
  "isCelsius": false,
  "weatherApiKey": "e343e791bbf3ec21",
}
`;
